provider/abstract_core.rb

```diff
@@ -21,6 +21,8 @@ class AbstractCore < Provider::Core
     # rubocop:disable Layout/EmptyLineBetweenDefs
     def generate_resource(data) end
     def generate_resource_tests(data) end
+    # Generates IAM policy support for the resource
+    def generate_iam_policy(data) end
```

Comment:
Should this be its own method, or do Terraform/Ansible expect to add IAM policy logic to the same files as the general resource handling? In InSpec I create separate files for IAM policies, so they can be handled within different methods, but I could imagine one of the providers combining them in the same file

Reply:
I would expect most of our IAM policies to be generated as separate resources. This is how they are almost all currently managed.

Reply:
It being it's own method makes sense, I see it as similar to https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/terraform.rb#L171-L188 or https://github.com/GoogleCloudPlatform/magic-modules/blob/master/provider/terraform.rb#L144-L169 in Terraform where we feature-detect to generate the file.

At a higher level, I don't think we should create a unified resource+iam in any provider; managing them separately as a fine-grained is the expectation from the API and the usage pattern we've already adopted in Terraform and now InSpec.

Reply:
Do we want the ability to generate the IAM policy but NOT the resource via MM? I have a use case for this in InSpec, wondering if I should generalize it or not

Reply:
Seems like a reasonable use case to me. We could use it in cases like Compute instance IAM so 👍 on the Terraform side.